src/integrationTest/java/com/mongodb/hibernate/BasicCrudTests.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.cfg.JdbcSettings.JAKARTA_JDBC_URL;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Sorts;
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.Version;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import org.bson.BsonDocument;
+import org.hibernate.annotations.DynamicUpdate;
+import org.hibernate.annotations.OptimisticLockType;
+import org.hibernate.annotations.OptimisticLocking;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.SessionFactoryScopeAware;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(
+        annotatedClasses = {
+            BasicCrudTests.Book.class,
+            BasicCrudTests.BookWithEmbeddedField.class,
+            BasicCrudTests.BookOptimisticLockWithVersion.class,
+            BasicCrudTests.BookOptimisticLockWithoutVersion.class
+        })
+class BasicCrudTests implements SessionFactoryScopeAware {
```

Comment:
I refactored the previous `BasicInsertionTests` to accommodate all the `basic CRUD testing` cases. `hibernate-testing` library doesn't integrate with JUnit 5 features elegantly (e.g. canonical usage requires passing `SessionFactoryScope` as the only testing method parameter, which conflicts with `@ParameterizedTests` in the first place), but it provides `SessionFactoryScopeAware` to work around.